php/Terminus/Models/Environment.php

```diff
@@ -407,6 +407,47 @@ public function getConnectionMode() {
   }
 
   /**
+   * Retrieves recommended DNS settings for this environment
+   *
+   * @return array[]
+   * @throws TerminusException
+   */
+  public function getDns() {
+    if ($this->site->get('service_level') == 'free') {
+      throw new TerminusException(
+        'To add custom domains, this site must be upgraded to a paid plan.',
+        [],
+        1
+      );
+    }
+    $response = $this->request->simpleRequest(
+      sprintf(
+        'sites/%s/environments/%s?environment_state=true',
+        $this->site->get('id'),
+        $this->get('id')
+      )
+    );
+    $hosts          = $response['data']->urls;
+    $load_balancers = (array)$response['data']->loadbalancers;
+    $load_balancer  = array_shift($load_balancers);
+    $dns_settings   = [];
+    foreach ($hosts as $host) {
+      $dns_settings[] = [
+        'host' => $host,
+        'record_type' => 'A',
+        'value' => $load_balancer->ipv4,
+      ];
+      $dns_settings[] = [
+        'host' => $host,
+        'record_type' => 'AAAA',
+        'value' => $load_balancer->ipv6,
+      ];
+    }
```

Comment:
I made the change. From the docs, it seems the way to identify a base hostname is by looking for any member of the full set of hostnames that does not have a dot followed immediately by another member's full name as a substring of itself.